### PR TITLE
Fix better print binary

### PR DESCRIPTION
**Pull request by evilebottnawi**
:star: :star: :star: 

All new examples will be tested on prettier playground and all output same as prettier

### CODE REVIEW

File: tests/comments/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -153,17 +153,11 @@ $bool = /*Comment */ true /*Comment */ || /*Comment */ false /*Comment */;
...
-    ) ||
+    ((is_array($this->modelData) || $this->modelData instanceof \\Countable) &&
+        0 === count($this->modelData)) ||
     // traversables that are not countable
-    (
-        $this->modelData instanceof \\Traversable &&
-            0 === iterator_count($this->modelData)
-    );
+    ($this->modelData instanceof \\Traversable &&
+        0 === iterator_count($this->modelData));
```
Comment (evilebottnawi) [inline tests/comments/__snapshots__/jsfmt.spec.js.snap]: Looks great :+1:

File: tests/retif/__snapshots__/jsfmt.spec.js.snap [hunk 2]
```diff
@@ -63,10 +63,8 @@ return ($testReallyReallyLong >= 1)
     ? $someOtherReallyReallyLongVariable
     : $thisOtherVariable;
 return true ?: 2;
-return (
-    $testReallyReallyLong === $someOtherReallyReallyLongVariable &&
-        $otherTest != $someOtherTest
-)
+return ($testReallyReallyLong === $someOtherReallyReallyLongVariable &&
+    $otherTest != $someOtherTest)
```
Comment (evilebottnawi) [inline tests/retif/__snapshots__/jsfmt.spec.js.snap]: as prettier

File: tests/bin/__snapshots__/jsfmt.spec.js.snap [hunk 3]
```diff
@@ -166,6 +244,196 @@ $value =
...
+}
+if (
+    $someReallyReallyReallyLongBooleanVariable +
+    $someReallyReallyReallyLongBooleanVariable % 100
+) {
+}
+if (
+    $someReallyReallyReallyLongBooleanVariable %
+        $someReallyReallyReallyLongBooleanVariable +
+    100
```
Comment (evilebottnawi) [inline tests/bin/__snapshots__/jsfmt.spec.js.snap]: Before we don't add indent and some cases looks very ugly

File: src/util.js [hunk 4]
```diff
@@ -429,6 +445,14 @@ function isProgramLikeNode(node) {
   return ["program", "declare", "namespace"].includes(node.kind);
 }
 
+function getNodeKindIncludeLogical(node) {
```
Comment (czosel) [inline src/util.js]: I'm not so happy with the name of this function, but after reading what it does it seems hard to come up with a better name. Is the naming from prettier core?

File: tests/comments/__snapshots__/jsfmt.spec.js.snap (hunk 1 above)
Comment (czosel) [reply tests/comments/__snapshots__/jsfmt.spec.js.snap]: :100:

**Comment by czosel** [comment]
:+1: Awesome work!

File: src/util.js (hunk 4 above)
Comment (evilebottnawi) [reply src/util.js]: @czosel no, `babylon` separate operators on `bin` and `logical`, in php all operators are `bin`

Comment (evilebottnawi) [reply src/util.js]: Anyway if somebody can find better name - PR welcome. We can change name in any time.

Comment (czosel) [reply src/util.js]: Ok... a minor improvement might be `getNodeKindIncludingLogical` and a little comment saying pretty much what you just commented?

**Pull request merged**